azurerm/internal/services/web/resource_arm_app_service_hybrid_connection.go

```diff
@@ -0,0 +1,205 @@
+package web
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+
+	"github.com/hashicorp/go-azure-helpers/response"
+
+	"github.com/Azure/azure-sdk-for-go/services/web/mgmt/2018-02-01/web"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmAppServiceHybridConnection() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAppServiceHybridConnectionCreateUpdate,
+		Read:   resourceArmAppServiceHybridConnectionRead,
+		Update: resourceArmAppServiceHybridConnectionCreateUpdate,
+		Delete: resourceArmAppServiceHybridConnectionDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"app_service_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAppServiceName,
+			},
+			"resource_group_name": azure.SchemaResourceGroupName(),
+			"relay_arm_uri": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+			"namespace_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"relay_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
```

Comment:
Could we move all computed properties to the bottom? generally we go required -> optional -> computed